cheroot/server.py

```diff
@@ -1773,20 +1762,93 @@ def error_log(self, msg='', level=20, traceback=False):
 
     def bind(self, family, type, proto=0):
         """Create (or recreate) the actual socket object."""
-        self.socket = socket.socket(family, type, proto)
-        prevent_socket_inheritance(self.socket)
+        sock = self.prepare_socket(
+            self.bind_addr,
+            family, type, proto,
+            self.nodelay, self.ssl_adapter,
+        )
+        sock = self.socket = self.bind_socket(sock, self.bind_addr)
+        self.bind_addr = self.resolve_real_bind_addr(sock)
+        return sock
+
+    def bind_unix_socket(self, bind_addr):
+        """Create (or recreate) a UNIX socket object."""
+        if IS_WINDOWS:
+            """
+            Trying to access socket.AF_UNIX under Windows
+            causes an AttributeError.
+            """
+            raise ValueError(  # or RuntimeError?
+                'AF_UNIX sockets are not supported under Windows.'
+            )
+
+        fs_permissions = 0o777  # TODO: allow changing mode
```

Comment:
I'm still uneasy with this as a default. I think the default should be not to change any permissions from whatever the OS sets.

Reply:
All servers I can recall have 0o666 as default permissions.
Also, take into account that if the permissions don't have rw, then nobody will be able to use this socket and our goal is to have it accessible by default. See `man -s 7 unix` for details:
```

   Pathname socket ownership and permissions
       In the Linux implementation, pathname sockets honor the permissions of the directory they are in.  Creation of a new
       socket fails if the process does not have write and search (execute) permission on the directory in which the socket
       is created.

       On Linux, connecting to a stream socket object requires write permission on that socket; sending  a  datagram  to  a
       datagram  socket  likewise  requires  write  permission on that socket.  POSIX does not make any statement about the
       effect of the permissions on a socket file, and on some systems (e.g.,  older  BSDs),  the  socket  permissions  are
       ignored.  Portable programs should not rely on this feature for security.

       When creating a new socket, the owner and group of the socket file are set according to the usual rules.  The socket
       file has all permissions enabled, other than those that are turned off by the process umask(2).

       The owner, group, and permissions of a pathname socket can be changed (using chown(2) and chmod(2)).
```